src/MSBuild/LiveLogger/TerminalBuffer.cs

```diff
@@ -100,10 +101,14 @@ public static void Render()
             }
 
             ShouldRerender = false;
+            ANSIBuilder.Formatting.ForegroundColor desiredColor =
+                topBarColor == TopBarColor.Error ? ANSIBuilder.Formatting.ForegroundColor.Red :
+                topBarColor == TopBarColor.Warning ? ANSIBuilder.Formatting.ForegroundColor.Yellow :
+                ANSIBuilder.Formatting.ForegroundColor.White;
```

Comment:
This totally works and you don't have to change it but I think I'd find this clearer with a pattern match:

```c#
(hasLoggedErrors, hasLoggedWarnings) switch {
    (true, _) => ConsoleColor.Red,
    (false, true) => ConsoleColor.Yellow,
    _ => ConsoleColor.White
}
```